src/nvapi_vulkan.cpp

```diff
@@ -0,0 +1,255 @@
+#include "nvapi_globals.h"
+#include "nvapi/nvapi_vulkan_low_latency_device.h"
+#include "util/util_statuscode.h"
+
+#include <libloaderapi.h>
+
+using namespace dxvk;
+
+static bool TryInitVkLowLatency() {
+    if (vkLowLatency && vkLowLatency->IsAvailable())
+        return true;
+
+    if (!resourceFactory)
+        return false;
+
+    vkLowLatency = resourceFactory->CreateVulkan("vulkan-1.dll");
+
+    if (vkLowLatency && vkLowLatency->IsAvailable())
+        return true;
+
+    vkLowLatency = resourceFactory->CreateVulkan("winevulkan.dll");
+
+    return vkLowLatency && vkLowLatency->IsAvailable();
+}
+
+extern "C" {
+    NvAPI_Status __cdecl NvAPI_Vulkan_InitLowLatencyDevice(HANDLE vkDevice, HANDLE* signalSemaphoreHandle) {
+        constexpr auto n = __func__;
+
+        if (log::tracing())
+            log::trace(n, log::fmt::ptr(vkDevice), log::fmt::ptr(signalSemaphoreHandle));
+
+        static std::once_flag initialized{};
+        std::call_once(initialized, TryInitVkLowLatency);
+
+        if (!vkLowLatency || !vkLowLatency->IsAvailable())
+            return NotSupported(n);
+
+        auto device = reinterpret_cast<VkDevice>(vkDevice);
+        auto semaphore = reinterpret_cast<VkSemaphore*>(signalSemaphoreHandle);
+
+        if (!device)
+            return InvalidArgument(n);
+
+        if (!semaphore)
+            return InvalidPointer(n);
+
+        auto [lowLatencyDevice, vr] = NvapiVulkanLowLatencyDevice::GetOrCreate(device, vkLowLatency->GetVkGetDeviceProcAddr());
+
+        if (!lowLatencyDevice)
+            return Error(n, vr);
+
+        *semaphore = lowLatencyDevice->GetSemaphore();
+
+        return Ok(n);
+    }
+
+    NvAPI_Status __cdecl NvAPI_Vulkan_DestroyLowLatencyDevice(HANDLE vkDevice) {
+        constexpr auto n = __func__;
+
+        if (log::tracing())
+            log::trace(n, log::fmt::ptr(vkDevice));
+
+        auto device = reinterpret_cast<VkDevice>(vkDevice);
+
+        if (!device)
+            return InvalidArgument(n);
+
+        return NvapiVulkanLowLatencyDevice::Destroy(device) ? Ok(n) : HandleInvalidated(n);
+    }
+
+    NvAPI_Status __cdecl NvAPI_Vulkan_GetSleepStatus(HANDLE vkDevice, NV_VULKAN_GET_SLEEP_STATUS_PARAMS* pGetSleepStatusParams) {
+        constexpr auto n = __func__;
+        thread_local bool alreadyLoggedOk = false;
+
+        if (log::tracing())
+            log::trace(n, log::fmt::ptr(vkDevice), log::fmt::nv_vk_get_sleep_status_params(pGetSleepStatusParams));
+
+        auto device = reinterpret_cast<VkDevice>(vkDevice);
+
+        if (!device)
+            return InvalidArgument(n);
+
+        if (!pGetSleepStatusParams)
+            return InvalidPointer(n);
+
+        if (pGetSleepStatusParams->version != NV_VULKAN_GET_SLEEP_STATUS_PARAMS_VER1)
+            return IncompatibleStructVersion(n, pGetSleepStatusParams->version);
+
+        auto lowLatencyDevice = NvapiVulkanLowLatencyDevice::Get(device);
+
+        // TODO: check native behavior for this case
```

Comment:
is this TODO for later of for this PR?

Reply:
Thanks and fine ofc to keep the todo en chack later. Since the impact is probably zero I would be ok with making the code active now. So: 

```cpp
        // TODO: check native behavior for this case
        if (!lowLatencyDevice)
            return HandleInvalidated(n);
```